packages/expo-image/ios/EXImageView.m

```diff
@@ -17,9 +36,67 @@ - (void)dealloc
 
 - (void)setSource:(NSDictionary *)source
 {
-  NSURL *imageURL = [RCTConvert NSURL:source[sourceUriKey]];
+  _source = source;
+  _needsReload = YES;
```

Comment:
```suggestion
  // TODO: Implement equatable image source abstraction
  _needsReload = YES;
```
? 😃

Reply:
Totally :) I felt this was out-of-scope for this PR, that's why I didn't :)